src/util.ts

```diff
@@ -376,10 +376,21 @@ export function flatAccessWithDatum(path: string, datum: 'datum' | 'parent' | 'd
  */
 export function replacePathInField(path: string) {
   return `${splitAccessPath(path)
-    .map(p => p.replace('.', '\\.'))
+    .map(p => replaceAll(p, '.', '\\.'))
     .join('\\.')}`;
 }
 
+/**
+ * Replace all ocurrences of a string with another string.
+ *
+ * @param string the string to replace in
+ * @param find the string to replace
+ * @param replace the replacement
+ */
+export function replaceAll(string: string, find: string, replace: string) {
+  return string.replace(new RegExp(find.replace(/[-/\\^$*+?.()|[\]{}]/g, '\\$&'), 'g'), replace);
```

Comment:
@domoritz why do we need a RegExp in this replacement?

Reply:
We need to replace every occurrence, not just the first one. However, normal strings need to be escaped.

Reply:
I see